src/wazuh_modules/wm_syscollector.c

```diff
@@ -38,15 +38,29 @@ syscollector_sync_message_func syscollector_sync_message_ptr = NULL;
 
 
 int queue_fd = 0;                                   // Output queue file descriptor
+bool in_shutdown = false;
+static pthread_mutex_t shutdown_mutex;
 
 static void wm_sys_send_diff_message(const void* data) {
     const int eps = 1000000/wm_max_eps;
-    wm_sendmsg(eps,queue_fd, data, WM_SYS_LOCATION, SYSCOLLECTOR_MQ);
+    pthread_mutex_lock(&shutdown_mutex);
```

Comment:
move the lock, read of the booleand and the unlock to a small static function that returns the value.